fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -481,16 +485,18 @@ ThreadFuture<Void> ThreadSafeTransaction::commit() {
 }
 
 Version ThreadSafeTransaction::getCommittedVersion() {
-	// This should be thread safe when called legally, but it is fragile
-	return tr->getCommittedVersion();
+	ISingleThreadTransaction* tr = this->tr;
+	return onMainThread([tr]() -> Future<Version> { return tr->getCommittedVersion(); }).get();
 }
 
 VersionVector ThreadSafeTransaction::getVersionVector() {
-	return tr->getVersionVector();
+	ISingleThreadTransaction* tr = this->tr;
+	return onMainThread([tr]() -> Future<VersionVector> { return tr->getVersionVector(); }).get();
```

Comment:
This function and `getSpanContext` might be better to return futures rather than being synchronous.